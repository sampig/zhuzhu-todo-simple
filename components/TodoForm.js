import React, { useCallback, useRef, useState } from 'react'
import { Button, StyleSheet, TextInput, View } from 'react-native'
import i18n from 'util/i18n-utils'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#ffffff',
  },
  textInput: {
    flex: 1,
    borderWidth: 2,
    fontSize: 20,
    margin: 5,
    borderRadius: 5,
    padding: 5,
  },
})

/**
 * Component to display a form of adding to-do.
 * @param {function} addNewTodo - a function to add a new to-do item
 * @returns {node}              - React node
 * @constructor
 */
function TodoForm ({ addNewTodo }) {
  const [todoValue, setTodoValue] = useState('')
  const inputRef = useRef()

  const submitAdd = useCallback(() => {
    inputRef?.current.blur()
    addNewTodo(todoValue)
    setTodoValue('')
  }, [inputRef, addNewTodo, todoValue, setTodoValue])

  return (
    <View style={styles.container}>
      <TextInput
        ref={inputRef}
        style={styles.textInput}
        value={todoValue}
        onChangeText={text => setTodoValue(text)}
        autoCapitalize='none'
        placeholder={i18n.t('enterTodo')}
      />
      <Button
        title='   +   '
        onPress={submitAdd}
      />
    </View>
  )
}

export default TodoForm
