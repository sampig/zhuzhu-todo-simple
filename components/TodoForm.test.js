import React from 'react'
import renderer from 'react-test-renderer'
import TodoForm from './TodoForm'

describe('The TodoForm component', () => {
  it('renders without crashing', () => {
    const render = renderer.create(<TodoForm addNewTodo={jest.fn()} />).toJSON()
    expect(render).toMatchSnapshot()
  })
})
