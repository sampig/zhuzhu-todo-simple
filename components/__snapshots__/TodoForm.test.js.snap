// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TodoForm component renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flexDirection": "row",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="enter a to-do..."
    rejectResponderTermination={true}
    style={
      Object {
        "borderRadius": 5,
        "borderWidth": 2,
        "flex": 1,
        "fontSize": 20,
        "margin": 5,
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
           +   
      </Text>
    </View>
  </View>
</View>
`;
